From bcb5409b23b4f558a6fcf7a1b86bd1118a147ff7 Mon Sep 17 00:00:00 2001
From: Yifeng Zhao <yifeng.zhao@rock-chips.com>
Date: Wed, 21 Oct 2020 18:19:26 +0800
Subject: [PATCH] drivers: mtd: nand: rockchip: fix a error while make

make error info: "error: passing argument 1 of 'nand_scan' from
incompatible pointer type [-Werror=incompatible-pointer-types]".

Fixes: 4c9539a31925 ("mtd: rawnand: Pass a nand_chip object to nand_scan()")
Change-Id: Ifb513a4f4466d64bfb88a220ba584220abedde8c
Signed-off-by: Yifeng Zhao <yifeng.zhao@rock-chips.com>
---
 drivers/mtd/nand/raw/rockchip_nand_v6.c | 6 +++---
 drivers/mtd/nand/raw/rockchip_nand_v9.c | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/mtd/nand/raw/rockchip_nand_v6.c b/drivers/mtd/nand/raw/rockchip_nand_v6.c
index 5ff2063a214d..09b7081ff5e5 100644
--- a/drivers/mtd/nand/raw/rockchip_nand_v6.c
+++ b/drivers/mtd/nand/raw/rockchip_nand_v6.c
@@ -748,14 +748,14 @@ static int rk_nand_chip_init(struct device *dev, struct rk_nfc *nfc,
 	mtd->dev.parent = dev;
 	mtd->name = "rk-nand";
 
-	ret = nand_scan(mtd, nsels);
+	ret = nand_scan(nand, nsels);
 	if (ret)
 		return ret;
 
 	ret = mtd_device_register(mtd, NULL, 0);
 	if (ret) {
 		dev_err(dev, "failed to register mtd device: %d\n", ret);
-		nand_release(mtd);
+		nand_release(nand);
 		return ret;
 	}
 
@@ -794,7 +794,7 @@ static void rk_nand_chips_cleanup(struct rk_nfc *nfc)
 	while (!list_empty(&nfc->chips)) {
 		chip = list_first_entry(&nfc->chips, struct rk_nand_chip,
 					node);
-		nand_release(nand_to_mtd(&chip->nand));
+		nand_release(&chip->nand);
 		list_del(&chip->node);
 	}
 }
diff --git a/drivers/mtd/nand/raw/rockchip_nand_v9.c b/drivers/mtd/nand/raw/rockchip_nand_v9.c
index bcbbf64f674f..62419c241e81 100644
--- a/drivers/mtd/nand/raw/rockchip_nand_v9.c
+++ b/drivers/mtd/nand/raw/rockchip_nand_v9.c
@@ -733,14 +733,14 @@ static int rk_nand_chip_init(struct device *dev, struct rk_nfc *nfc,
 	mtd->dev.parent = dev;
 	mtd->name = "rk-nand";
 
-	ret = nand_scan(mtd, nsels);
+	ret = nand_scan(nand, nsels);
 	if (ret)
 		return ret;
 
 	ret = mtd_device_register(mtd, NULL, 0);
 	if (ret) {
 		dev_err(dev, "failed to register mtd device: %d\n", ret);
-		nand_release(mtd);
+		nand_release(nand);
 		return ret;
 	}
 
@@ -779,7 +779,7 @@ static void rk_nand_chips_cleanup(struct rk_nfc *nfc)
 	while (!list_empty(&nfc->chips)) {
 		chip = list_first_entry(&nfc->chips, struct rk_nand_chip,
 					node);
-		nand_release(nand_to_mtd(&chip->nand));
+		nand_release(&chip->nand);
 		list_del(&chip->node);
 	}
 }
-- 
2.35.3

